internal/controller/dpuoperatorconfig_controller_test.go

```diff
@@ -60,193 +72,259 @@ func createDpuOperatorNameSpace() *corev1.Namespace {
 	return namespace
 }
 
-func ensureDpuOperatorNamespace() {
-	By("create test DPU operator namespace")
-	namespace := createDpuOperatorNameSpace()
-	Expect(k8sClient.Create(context.Background(), namespace)).Should(Succeed())
+func dpuOperatorCR(name string, mode string, ns *corev1.Namespace) *configv1.DpuOperatorConfig {
+	config := &configv1.DpuOperatorConfig{}
+	config.SetNamespace(ns.Name)
+	config.SetName(name)
+	config.Spec = configv1.DpuOperatorConfigSpec{
+		Mode:     mode,
+		LogLevel: 2,
+	}
+	return config
+}
+
+func createNameSpace(client client.Client, ns *v1.Namespace) {
+	// ignore error when createing the namespace since it can already exist
+	client.Create(context.Background(), ns)
+	found := v1.Namespace{}
+	err := client.Get(context.Background(), types.NamespacedName{Namespace: testNamespace, Name: ns.GetName()}, &found)
```

Comment:
We assume that namespaces are created instantly here. Would this always be the case? I expect similar code from `deleteNameSpace`

Reply:
Good point